java/base-client/src/main/java/no/bankaxept/epayment/client/base/accesstoken/AccessToken.java

```diff
@@ -27,21 +36,41 @@ public Instant getExpiry() {
         return expiry;
     }
 
-    public long millisUntilTenMinutesBeforeExpiry(Clock clock) {
-        return Duration.between(clock.instant().plusSeconds(600), expiry).toMillis();
+    public Integer getExpirySecondsFromStart() {
+        return expirySecondsFromStart;
+    }
+
+    public String getInput() {
+        return input;
+    }
+
+
+    public long tenSecondsBeforeExpiry(Clock clock) {
+        if(expirySecondsFromStart != null) {
+            return (expirySecondsFromStart * 1000) - 1000;
+        }
+        return Duration.between(clock.instant(), expiry.minusSeconds(10)).toMillis();
     }
 
     private static class Parser {
-        private static final Pattern tokenPattern = Pattern.compile("\"accessToken\"\\s*:\\s*\"(.*)\"");
-        private static final Pattern expiryPattern = Pattern.compile("\"expiresOn\"\\s*:\\s*(\\d+)");
+        private static final Pattern tokenPattern = Pattern.compile("\"access_token\"\\s*:\\s*\"(.*?)\"");
+        private static final Pattern expiryPattern = Pattern.compile("\"expires_on\"\\s*:\\s*(\\d+)");
+        private static final Pattern expirySecondsFromStartPattern = Pattern.compile("\"expires_in\"\\s*:\\s*(\\d+)");
 
-        static AccessToken parse(String input) {
+        static AccessToken parse(String input, Clock clock) {
             var tokenMatcher = tokenPattern.matcher(input);
             var expiryMatcher = expiryPattern.matcher(input);
-            if (!tokenMatcher.find() || !expiryMatcher.find()) {
-                throw new IllegalArgumentException("Could not parse token or expiry"); //onError
+            var expirySecondsFromStartMatcher = expirySecondsFromStartPattern.matcher(input);
+            if ( !tokenMatcher.find()) {
+                throw new IllegalArgumentException("Could not parse token: " + input);
```

Comment:
For debugging it was nice to output the whole token.. But i guess this could be bad to log, someone could technically use it?

Reply:
I removed input from the exception agai